lib/stripe/subscriptions/coupon.ex

```diff
@@ -57,19 +55,19 @@ defmodule Stripe.Coupon do
   @doc """
   Create a coupon.
   """
-  @spec create(params, Stripe.options) :: {:ok, t} | {:error, Stripe.Error.t}
+  @spec create(params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
         when params: %{
-               id: String.t,
+               id: String.t(),
                duration: :forever | :once | :repeating,
                amount_off: pos_integer,
-               currency: String.t,
+               currency: String.t(),
                duration_in_months: pos_integer,
                max_redemptions: pos_integer,
                metadata: %{
-                 optional(String.t) => String.t
+                 optional(String.t()) => String.t()
                },
                percent_off: pos_integer,
-               redeem_by: Stripe.timestamp
+               redeem_by: Stripe.timestamp()
```

Comment:
My guess is that a number of the `when params` here are actually optional. You can look at some of my recent PRs to see how I've handled that.